satpy/tests/reader_tests/test_modis_l1b.py

```diff
@@ -0,0 +1,306 @@
+                           )
+    inst_metadata = "GROUP = ASSOCIATEDPLATFORMINSTRUMENTSENSOR\n\n"\
+                    "OBJECT = ASSOCIATEDPLATFORMINSTRUMENTSENSORCONTAINER\nCLASS = \"1\"\n\n" \
+                    "OBJECT = ASSOCIATEDSENSORSHORTNAME\nCLASS = \"1\"\nNUM_VAL = 1\n" \
+                    "VALUE = \"MODIS\"\nEND_OBJECT = ASSOCIATEDSENSORSHORTNAME\n\n" \
+                    "OBJECT = ASSOCIATEDPLATFORMSHORTNAME\nCLASS = \"1\"\nNUM_VAL = 1\n" \
+                    "VALUE = \"Terra\"\nEND_OBJECT = ASSOCIATEDPLATFORMSHORTNAME\n\n" \
+                    "OBJECT = ASSOCIATEDINSTRUMENTSHORTNAME\nCLASS = \"1\"\nNUM_VAL = 1\n" \
+                    "VALUE = \"MODIS\"\nEND_OBJECT = ASSOCIATEDINSTRUMENTSHORTNAME\n\n" \
+                    "END_OBJECT = ASSOCIATEDPLATFORMINSTRUMENTSENSORCONTAINER\n\n" \
+                    "END_GROUP              = ASSOCIATEDPLATFORMINSTRUMENTSENSOR\n\n"
+    collection_metadata = "GROUP = COLLECTIONDESCRIPTIONCLASS\n\nOBJECT = SHORTNAME\nNUM_VAL = 1\n"\
+                          "VALUE = \"MOD021KM\"\nEND_OBJECT = SHORTNAME\n\n"\
+                          "OBJECT = VERSIONID\nNUM_VAL = 1\nVALUE = 6\nEND_OBJECT = VERSIONID\n\n"\
+                          "END_GROUP = COLLECTIONDESCRIPTIONCLASS\n\n"
+    core_metadata_header += "\n\n" + inst_metadata + collection_metadata
+    struct_metadata_header = "GROUP=SwathStructure\n"\
+                             "GROUP=SWATH_1\n"\
+                             "GROUP=DimensionMap\n"\
+                             "OBJECT=DimensionMap_2\n"\
+                             "GeoDimension=\"2*nscans\"\n"\
+                             "END_OBJECT=DimensionMap_2\n"\
+                             "END_GROUP=DimensionMap\n"\
+                             "END_GROUP=SWATH_1\n"\
+                             "END_GROUP=SwathStructure\nEND"
+    archive_metadata_header = "GROUP = ARCHIVEDMETADATA\nEND_GROUP = ARCHIVEDMETADATA\nEND"
+    setattr(h, 'CoreMetadata.0', core_metadata_header)  # noqa
+    setattr(h, 'StructMetadata.0', struct_metadata_header)  # noqa
+    setattr(h, 'ArchiveMetadata.0', archive_metadata_header)  # noqa
+
+    # Fill datasets
+    for dataset in TEST_DATA:
+        v = h.create(dataset, TEST_DATA[dataset]['type'], TEST_DATA[dataset]['data'].shape)
+        v[:] = TEST_DATA[dataset]['data']
+        dim_count = 0
+        for dimension_name in TEST_DATA[dataset]['attrs']['dim_labels']:
+            v.dim(dim_count).setname(dimension_name)
+            dim_count += 1
+        v.setfillvalue(TEST_DATA[dataset]['fill_value'])
+        v.scale_factor = TEST_DATA[dataset]['attrs'].get('scale_factor', SCALE_FACTOR)
+        for attr_key, attr_val in TEST_DATA[dataset]['attrs'].items():
+            if attr_key == 'dim_labels':
+                continue
+            setattr(v, attr_key, attr_val)
+    h.end()
+    return base_dir, file_name
+
+
+class TestModisL2(unittest.TestCase):
```

Comment:
This class should be moved to a dedicated test file or alternatively this file should be renamed.

Reply:
This is actually just misnamed since it was copy/pasted from the L2 tests. It should be TestModisL1b.